verifiers/groth16/src/groth16.rs

```diff
@@ -40,6 +41,18 @@ pub struct VerificationKeyWithCurve {
     pub gamma_abc_g1: Vec<G1>,
 }
 
+impl MaxEncodedLen for VerificationKeyWithCurve {
+    fn max_encoded_len() -> usize {
+        let g1_size: u32 = G1::max_encoded_len()
+            .try_into()
+            .expect("Should be a valid u32. qed");
+        Curve::max_encoded_len()
+            + G1::max_encoded_len()
+            + 3 * G2::max_encoded_len()
+            + vec_max_encoded_len(crate::MAX_NUM_INPUTS * g1_size)
```

Comment:
This should actually be
```rust
(crate::MAX_NUM_INPUTS + 1) * g1_size
```
You could add a test which builds a vk of maximum size, encodes it, and then checks that its actual size coincides with the one returned by `max_encoded_len()`

Reply:
DOne.... and thx! I found a bug, ok minor because I been overestimating it but a bug !!!